lightning/restore/restore.go

```diff
@@ -339,19 +348,42 @@ func (rc *RestoreController) listenCheckpointUpdates(wg *sync.WaitGroup) {
 
 		lock.Unlock()
 
+		// Note: about gofailLabel1, gofailLabel2, etc.., just a hack way
+		// for avoiding generated error message e.g:
+		// `failpoint: "github.com/pingcap/tidb-lightning/lightning/restore/FailIfIndexEngineImported" got value 140 of type "int" but expected type "int"`
+
 		// gofail: var FailIfImportedChunk struct{}
 		// if _, ok := scp.merger.(*ChunkCheckpointMerger); ok {
 		// 	wg.Wait()
 		// 	panic("forcing failure due to FailIfImportedChunk")
 		// }
-		// continue
+		// goto gofailLabel1
+	gofailLabel1:
+		if false {
+			goto gofailLabel1
+		}
 
 		// gofail: var FailIfStatusBecomes int
 		// if merger, ok := scp.merger.(*StatusCheckpointMerger); ok && merger.EngineID >= 0 && int(merger.Status) == FailIfStatusBecomes {
 		// 	wg.Wait()
 		// 	panic("forcing failure due to FailIfStatusBecomes")
 		// }
-		// continue
+		// goto gofailLabel2
+	gofailLabel2:
+		if false {
+			goto gofailLabel2
+		}
+
+		// gofail: var FailIfIndexEngineImported int
```

Comment:
it seems failpoint `FailIfStatusBecomes` is able to take place of `FailIfIndexEngineImported `

Reply:
We need an additional failpoint to cover FailIfIndexEngineImported also. See `tests/checkpoint`